crates/block-producer/src/rpc_client.rs

```diff
@@ -765,6 +766,68 @@ impl RPCClient {
         Ok(collected)
     }
 
+    pub async fn query_custodian_type_script(
+        &self,
+        sudt_script_hash: [u8; 32],
+    ) -> Result<Option<Script>> {
+        let rollup_context = &self.rollup_context;
+
+        let custodian_lock = Script::new_builder()
+            .code_hash(rollup_context.rollup_config.custodian_script_type_hash())
+            .hash_type(ScriptHashType::Type.into())
+            .args(rollup_context.rollup_script_hash.as_slice().pack())
+            .build();
+
+        let search_key = SearchKey {
+            script: ckb_types::packed::Script::new_unchecked(custodian_lock.as_bytes()).into(),
+            script_type: ScriptType::Lock,
+            filter: None,
+        };
+        let order = Order::Desc;
+        let limit = Uint32::from(DEFAULT_QUERY_LIMIT as u32);
+
+        let mut sudt_script = None;
+        let mut cursor = None;
+
+        while sudt_script.is_none() {
+            let cells: Pagination<Cell> = to_result(
+                self.indexer_client
+                    .request(
+                        "get_cells",
+                        Some(ClientParams::Array(vec![
+                            json!(search_key),
+                            json!(order),
+                            json!(limit),
+                            json!(cursor),
+                        ])),
+                    )
+                    .await?,
+            )?;
+
+            if cells.last_cursor.is_empty() {
+                return Ok(None);
+            }
+            cursor = Some(cells.last_cursor);
+
+            for cell in cells.objects.into_iter() {
+                let sudt_type_script = match cell.output.type_.clone() {
```

Comment:
You should also verify the lock.args which is not verified on-chain, an attacker can break a godwoken node by fill an invalid args. Pls check the `sanitize_deposit_cells` function.

Reply:
The result script is only used to calculate estimate withdrawal cell's capacity in ```minimal_capacity_verifier```. I think it's better move it to ```minimal_capacity_verifier()```.

Reply:
Now it's verified. Done